### PR TITLE
Clean up the fetched tar if the download times out

### PR DESCRIPTION
**Pull request by ryndaniels**

### CODE REVIEW

File: bin/casher [hunk 1]
```diff
@@ -72,10 +73,14 @@ class Casher
       end
     end
     unless archive_found
-      msg "could not download cache", :red
-      if File.exist? @fetch_tar
-        rm @fetch_tar
-      end
+      cleanup_cache
```
Comment (meatballhat) [inline bin/casher]: Given how small the body of the `unless` block is now, I would switch this to a trailing condition a la `cleanup_cache unless archive_found`.

File: spec/casher_spec.rb [hunk 2]
```diff
@@ -23,4 +23,16 @@
       subject.run('fetch', tgz_url, tbz_url)
     end
   end
+
+  context 'when the curl call times out' do
+    before :each do
+      expect(subject).to receive(:system).with(/curl\b.*#{tgz_url.gsub('?','\?')}/m).and_raise(TimeoutError)
+    end
+
+    it 'does not clean up the fetch_tar' do
```
Comment (meatballhat) [inline spec/casher_spec.rb]: Now that this works, should the description be `'cleans up the fetch_tar'`? :grin:

Comment (ryndaniels) [reply spec/casher_spec.rb]: AHA good catch! :D :D :D

File: bin/casher [hunk 3]
```diff
@@ -71,11 +72,13 @@ class Casher
...
-      if File.exist? @fetch_tar
-        rm @fetch_tar
-      end
+    cleanup_cache unless archive_found
+  end
+
+  def cleanup_cache
+    msg "could not download cache", :red
+    if File.exist? @fetch_tar
+      rm @fetch_tar
```
Comment (meatballhat) [inline bin/casher]: This isn't specific to the PR, but I don't know why we wouldn't use `FileUtils.rm_f(@fetch_tar)`.  Tell-don't-ask and all that jazz :saxophone: :man_shrugging:

**Pull request merged**